src/main/java/com/gmail/goosius/siegewar/playeractions/PlaceBlock.java

```diff
@@ -57,6 +60,92 @@ public static void evaluateSiegeWarPlaceBlockRequest(Player player, Block block,
 			if (!TownyAPI.getInstance().getTownyWorld(block.getWorld()).isWarAllowed())
 				return;
 
+			//If the event has already been cancelled by Towny...
+			if(event.isCancelled()) {
+				if(!SiegeWarSettings.isWallBreachingEnabled())
+					return; //Without wall breaching, SW doesn't un-cancel events
+				Town town = TownyAPI.getInstance().getTown(block.getLocation());
+				if(town == null)
+					return; //SW currently doesn't un-cancel wilderness events
+				if(!SiegeController.hasActiveSiege(town))
+					return; //SW doesn't un-cancel events in unsieged towns
+					
+				//Ensure player is on the town-hostile siege side				
+				Resident resident = TownyAPI.getInstance().getResident(player);
+				if(resident == null)
+					return;
+				if(!resident.hasNation())
+					return;
+				Siege siege = SiegeController.getSiege(town);
+				SiegeSide playerSiegeSide = SiegeWarAllegianceUtil.calculateCandidateSiegePlayerSide(player, resident.getTownOrNull(), siege);
+				if(playerSiegeSide == SiegeSide.NOBODY)
+					return; 
+				switch(siege.getSiegeType()) {
+					case CONQUEST:
+					case SUPPRESSION:
+						if(playerSiegeSide != SiegeSide.ATTACKERS)
+							return;
+						break;
+					case REVOLT:
+					case LIBERATION:
+						if(playerSiegeSide != SiegeSide.DEFENDERS)
+							return;
+						break;
+				}
+
+				//Ensure player has permission
+				if (!TownyUniverse.getInstance().getPermissionSource().testPermission(event.getPlayer(), SiegeWarPermissionNodes.SIEGEWAR_NATION_SIEGE_USE_BREACH_POINTS.getNode())) {
+					event.setMessage(Translation.of("msg_err_action_disable"));
+					return; 
+				}
```

Comment:
This could potentially be moved up higher, near to the if (event.isCancelled()) and do away with the event.setMessage().

Reply:
- I see what you mean.
- But that would mean that regular soldiers trying to break blocks would just get normal Towny "you cannot break blocks" message, instead of the different "permission denied". Maybe that difference might help support teams to diagnose and fix issues. 
- But I don't mind. So, move/don't-move?

Reply:
Will do as per previous comment

Reply:
Done